ifrs17/Utils/ActivityLog.ipynb

```diff
@@ -354,6 +354,160 @@
+                "\n                }",
+                "\n                catch (Exception)",
+                "\n                {",
+                "\n                    importSucceeded = false;",
+                "\n                    activity = activity with {SourceId = null};",
+                "\n                }",
+                "\n                activity = activity with {Category = \"Import from Data Set\"};",
+                "\n                if (importSucceeded) await DataSource.UpdateAsync<ImportDataSet>(importDataSet.RepeatOnce());",
+                "\n                break;",
+                "\n",
+                "\n            default:",
+                "\n                throw new Exception(\"Import Options object is not an instance of an appropriate class.\");",
+                "\n                break;",
+                "\n        }",
+                "\n        await DataSource.UpdateAsync<ImportExportActivity>(activity.RepeatOnce());",
+                "\n        await DataSource.CommitAsync();",
+                "\n        return log;",
+                "\n    }",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public record ExportBuilderWriter(DocumentBuilder Builder, ",
+                "\n                                ISessionVariable Session, ",
+                "\n                                IDataSource DataSource, ",
+                "\n                                IDataSetImportVariable ImportVariable)",
+                "\n{",
+                "\n    public async Task<ExportResult> ExecuteAsync()",
+                "\n    {",
+                "\n        var exportResult = await Builder.ExecuteAsync();",
+                "\n        var exportFile = await (new ExportFile(Builder, ImportVariable, Session)).InitializeExportDataAsync();",
+                "\n        var activity = new ImportExportActivity(exportResult.ActivityLog, Session) with {Category = \"Export to File\", ",
+                "\n                                                                                    SourceId = exportFile.Id};",
+                "\n        await DataSource.UpdateAsync<ExportFile>(exportFile.RepeatOnce());",
+                "\n        await DataSource.UpdateAsync<ImportExportActivity>(activity.RepeatOnce());",
+                "\n        await DataSource.CommitAsync();",
+                "\n        return exportResult;",
+                "\n    }",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public static ImportBuilderWriter WithLogsRecordedInDb(this ImportOptionsBuilder builder, ",
```

Comment:
It just bothers me the InDb in the namehere, as this would also work if you don't have a Db (say, in memory --> you would also create a log in the datasource).

What about just .WithLogRecords? You are anyway passing IDataSource as parameter, and this should make clear where the log records are being created.

Reply:
let's use WithActivityLog
Discussed with Daniel. 
Thanks